### DISCUSSION TITLE
Cannot add any thing to jump list

### DISCUSSION THREAD

**Discussion by richardvanbergen** [cbochs/grapple.nvim#152] [General]
Hey guys, I'm coming across a weird problem, I think it's an issue with the buffer for this particular git branch as grapple is working just fine in other branches and repos. I can't toggle any files with `Grapple toggle`.

```
Expected value but found T_END at character 1
```

I don't know if there's any way I could reproduce as I don't know how it got into this state. Is there a way I could "reset" it for this scope?

If there's anything I could provide to make it easier please let me know, I'm kind of a noob at nvim, I'm not sure where the diagnostic information for this could be found.

**Comment by cbochs** [comment]
Hi @richardvanbergen, maybe if you can provide a bit more information I can try and help you out here. 

- How do you have Grapple configured?
- When does this error appear? 
- Is it only when you call `Grapple toggle`?
- Does the error appear once? Does it appear multiple times?
- What is the git directory and git branch that you see these errors for?
- Do you know how to run lua code and get its output? If so, try the following:
    - run `require("grapple.app").get().settings.save_path`. Look there to see if there is a .json file that looks like the git directory and branch you are having issues with. If you find it, what are the contents?

**Reply by richardvanbergen** [reply]
Hi @cbochs thanks for taking the time to get back to me.

> How do you have Grapple configured?

I'm using kickstart.nvim and have it defined as a plugin file.

```lua
return {
  "cbochs/grapple.nvim",
  dependencies = {
    { "nvim-tree/nvim-web-devicons", lazy = true }
  },
  opts = {
    scope = "git_branch",
  },
  event = { "BufReadPost", "BufNewFile" },
  cmd = "Grapple",
  keys = {
    { "<leader>m", "<cmd>Grapple toggle<cr>",         desc = "Grapple toggle" },
    { "<leader>k", "<cmd>Grapple toggle_tags<cr>",    desc = "Grapple toggle tags" },
    { "<leader>K", "<cmd>Grapple toggle_scopes<cr>",  desc = "Grapple toggle scopes" },
    { "<leader>j", "<cmd>Grapple cycle forward<cr>",  desc = "Grapple cycle forward" },
    { "<leader>J", "<cmd>Grapple cycle backward<cr>", desc = "Grapple cycle backward" },
  },
}
```

> When does this error appear?

Whenever I run `Grapple toggle` or `Grapple cycle`

> Is it only when you call Grapple toggle?

Seems to affect cycle too.

> Does the error appear once? Does it appear multiple times?

I think once, I only get one message after the command it triggered.

> What is the git directory and git branch that you see these errors for?

I have to redact the names slightly, I'm guessing your asking to look for funny characters.

Git directory: /Users/richardvanbergen/code/mycompany-frontend-monorepo/.git
Branch: f/TK-123-client-pricing-script

I did just test for the `f/` prefix on the branch name but it seems to be OK.

Observation: **It only affects literally that branch. Everything else works fine.** This suggests to me that something has gotten corrupt. I did have an issue with files not being able to be written to the disk the other day because it was out of space which is when I started noticing this.

> :lua require("grapple.app").get().settings.save_path

Sorry if I got the command wrong. 🙈

Output: E5107: Error loading lua [string ":lua"]:1: '=' expected near '<eof>'

Command works until I try to access `.settings`

**Reply by cbochs** [reply]
> I have to redact the names slightly, I'm guessing your asking to look for funny characters.

No worries. That all looks fine.

> I did have an issue with files not being able to be written to the disk the other day because it was out of space which is when I started noticing this.

This is probably the issue. Hopefully you've freed up more space since then!

> `:lua require("grapple.app").get().settings.save_path`
> Sorry if I got the command wrong. 🙈
> Output: E5107: Error loading lua [string ":lua"]:1: '=' expected near ''

Instead, try this (put an `=` sign after `:lua`):

```lua
:lua=require("grapple.app").get().settings.save_path
```

That should output Grapple's save directory. If that doesn't work, then at the very least you should be able to do:

```lua
:lua=vim.fn.stdpath("data")
```

If you go to that location you should see `grapple/`, which is Grapple's save directory. Look there for a `.json` file that matches your git repo and git branch. I'm guessing the contents inside that file are corrupt (i.e. incomplete write, invalid JSON).

**Reply by cbochs** [reply]
@richardvanbergen Another approach would be to reset/delete the scope through the Loaded Scopes window. To do that, the following steps should work:

1. Open the Loaded Scopes windows with `:Grapple toggle_loaded`
2. Use `g.` to show all your scope files (loaded and unloaded). Alternatively, you could just open the window and show all with `:Grapple toggle_loaded all=true`
3. Find the offending scope that matches your git repo/git branch
4. Press (uppercase) `X` to reset/delete it

**Reply by richardvanbergen** [reply]
@cbochs Excellent, that's fixed the issue! I'm sorry I don't have the expertise to help you find the root cause but I seems like it's an edge-case anyway.

Thank you for unsticking me and for the great plugin. I use it every day. :)

**Reply by cbochs** [reply]
Great to hear (both that the above resolved your issue and that you enjoy Grapple)

> I'm sorry I don't have the expertise to help you find the root cause

No worries at all. Glad to have this issue raised (and solved) in the case others come across the same issue as you 🙂
